### PR TITLE
Add --ignoreDirectoryPattern cli option

### PR DESCRIPTION
**Pull request by levithomason**
Rebased version of the dead https://github.com/mikeal/watch/pull/93.

### CODE REVIEW

**Comment by vanderlee** [comment]
Let's hope it gets merged before the base changes again!

**Comment by levithomason** [comment]
No worries, I have rights to merge now and most certainly can rebase as needed ;)

File: cli.js [hunk 1]
```diff
@@ -35,6 +43,11 @@ if(argv.ignoreDotFiles || argv.d)
 if(argv.ignoreUnreadable || argv.u)
   watchTreeOpts.ignoreUnreadableDir = true
 
+if(argv.ignoreDirectoryPattern || argv.p) {
+  var match = (argv.ignoreDirectoryPattern || argv.p).match(/^\/(.*)\/([gimuy])$/);
```
Comment (levithomason) [inline cli.js]: #99 adds minimatch for matching, we should consider staying consistent.

**Comment by levithomason** [comment]
Rebased, merging.

**Comment by vanderlee** [comment]
Great :) When will the version get bumped so NPM sees it?

**Comment by levithomason** [comment]
Already done, v0.19.1 should be what you're looking for.

**Comment by vanderlee** [comment]
My bad; "watch" was indeed updated

**Pull request merged**